manager/controlapi/secret.go

```diff
@@ -51,6 +52,52 @@ func (s *Server) GetSecret(ctx context.Context, request *api.GetSecretRequest) (
 	return &api.GetSecretResponse{Secret: secret}, nil
 }
 
+// UpdateSecret updates a Secret referenced by SecretID with the given SecretSpec.
+// - Returns `NotFound` if the Secret is not found.
+// - Returns `InvalidArgument` if the SecretSpec is malformed or anything other than Labels is changed
+// - Returns an error if the update fails.
+func (s *Server) UpdateSecret(ctx context.Context, request *api.UpdateSecretRequest) (*api.UpdateSecretResponse, error) {
+	if request.SecretID == "" || request.SecretVersion == nil {
+		return nil, grpc.Errorf(codes.InvalidArgument, errInvalidArgument.Error())
+	}
+
+	var secret *api.Secret
+	err := s.store.Update(func(tx store.Tx) error {
+		secret = store.GetSecret(tx, request.SecretID)
+		if secret == nil {
+			return nil
+		}
+
+		// If there are no changes, return
+		if reflect.DeepEqual(secret.Spec, request.Spec) {
+			return nil
+		}
+
+		// We only allow changes to the Labels, so if there are no changes here
+		// something else changed, and we should reject it
+		if reflect.DeepEqual(secret.Spec.Annotations.Labels, request.Spec.Annotations.Labels) {
+			return grpc.Errorf(codes.InvalidArgument, "only updates to Labels are allowed")
+		}
```

Comment:
I don't see other controlapi `Update` functions rejecting updates that don't make any changes.